config/configLoadHelpers.go

```diff
@@ -108,17 +109,83 @@ func load(r io.Reader, format Format, into any) error {
 	}
 }
 
-func validateConfig(opts *CmdEnv) ([]string, error) {
-	location := opts.ConfigLocation
-	r, format, err := getReaderFor(location)
-	if err != nil {
-		return nil, err
+// This loads all the named configs into destination in the order they are listed.
+// It returns the MD5 hash of the collected configs as a string (if there's only one
+// config, this is the hash of that config; if there are multiple, it's the hash of
+// all of them concatenated together).
+func loadConfigsInto(dest any, locations []string) (string, error) {
+	// start a hash of the configs we read
+	h := md5.New()
+	for _, location := range locations {
+		// trim leading and trailing whitespace just in case
+		location := strings.TrimSpace(location)
+		r, format, err := getReaderFor(location)
+		if err != nil {
+			return "", err
+		}
+		defer r.Close()
+		// write the data to the hash as we read it
+		rdr := io.TeeReader(r, h)
+
+		// when working on a struct, load doesn't overwrite destination values
```

Comment:
nit - do you mind to change this comment from a double negative to something like `load only overwrite destination values that are specified in a file`